tests/python_package_test/test_dask.py

```diff
@@ -36,6 +44,102 @@ def listen_port():
 listen_port.port = 13000
 
 
+def r2_score(dy_true, dy_pred):
+    """Helper function taken from dask_ml.metrics: computes coefficient of determination."""
+    numerator = ((dy_true - dy_pred) ** 2).sum(axis=0)
+    denominator = ((dy_true - dy_pred.mean(axis=0)) ** 2).sum(axis=0)
+    return (1 - numerator / denominator).compute()
+
+
+def _make_ranking(n_samples=100, n_features=20, n_informative=5, gmax=1, random_gs=False, avg_gs=10, random_state=0):
```

Comment:
Could you add docs explaining what each of these parameters means? There are a lot to reason about here and I want to be sure that people editing or adding to these tests in the future understand what's going on.

I think it's:

* `n_samples` = total number of documents (records) in the dataset
* `n_features` = total number of features in the dataset
* `n_informative` = how many of the features should be informative. If this is greater than `n_features`, the dataset will have `n_features` features, and all of them will be informative. Features are guaranteed to be "informative" for ranking by creating their values as `bias + beta*(actual_relevance_scores)`, with a random `bias` and `beta`.
* `gmax` = Maximum relevance value. Relevance scores are integers. If you set this to 2, for example, all documents in a group will get relevant scores of either 0, 1, or 2.
* `random_gs` = True if you want group size to be random, False if you want all groups to be the same size
* `avg_gs` = number of documents (records) in each group

Reply:
Yep, absolutely. Didn't know how much docstrings in the test modules were emphasized. Heads up, I'm also going to add a `group` parameter that's analogous to the LightGBMRanker's `group` parameter, so that we can specify group sizes directly (as requested). When specified, the `group` vector will override `n_samples` and `avg_gs` and `#data = sum(group)`.

Reply:
ok awesome, thanks! In general we don't put a ton of emphasis on docstrings for functions in tests, but this one has a lot of parameters so I thought it would be good, to be sure future contributors understand how to use it

Reply:
Addressed in d6e5209